File: app/src/main/java/com/abishov/hexocat/common/utils/OnErrorHandler.java
package com.abishov.hexocat.common.utils;

import io.reactivex.exceptions.OnErrorNotImplementedException;
import io.reactivex.functions.Consumer;

public final class OnErrorHandler implements Consumer<Throwable> {

  private OnErrorHandler() {
    // use factory method
  }

  public static Consumer<Throwable> create() {
    return new OnErrorHandler();
  }

  @Override
  public void accept(Throwable throwable) {
    throw new OnErrorNotImplementedException(throwable);
  }
}

File: app/src/main/java/com/abishov/hexocat/common/utils/OnErrorHandler.kt
package com.abishov.hexocat.common.utils

import io.reactivex.exceptions.OnErrorNotImplementedException
import io.reactivex.functions.Consumer

class OnErrorHandler private constructor() : Consumer<Throwable> {

  override fun accept(throwable: Throwable) {
    throw OnErrorNotImplementedException(throwable)
  }

  companion object {
    fun create(): Consumer<Throwable> {
      return OnErrorHandler()
    }
  }
}
